src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/UniqueAccessCodeServiceImpl.java

```diff
@@ -44,88 +43,70 @@
     private CaseType caseCaseType;
 
-    private LinkingCombination(FormType uacFormType, CaseType uacCaseType, CaseType caseCaseType) {
+    private LinkingCombination(FormType uacFormType, CaseType caseCaseType) {
       this.uacFormType = uacFormType;
-      this.uacCaseType = uacCaseType;
       this.caseCaseType = caseCaseType;
     }
 
-    static Optional<LinkingCombination> lookup(
-        FormType uacFormType, CaseType uacCaseType, CaseType caseCaseType) {
+    static Optional<LinkingCombination> lookup(FormType uacFormType, CaseType caseCaseType) {
       return Stream.of(LinkingCombination.values())
-          .filter(
-              row ->
-                  row.uacFormType == uacFormType
-                      && row.uacCaseType == uacCaseType
-                      && row.caseCaseType == caseCaseType)
+          .filter(row -> row.uacFormType == uacFormType && row.caseCaseType == caseCaseType)
           .findAny();
     }
   }
 
   /** Constructor */
-  public UniqueAccessCodeServiceImpl() {
-
-    MapperFactory mapperFactory = new DefaultMapperFactory.Builder().build();
-    ConverterFactory converterFactory = mapperFactory.getConverterFactory();
-    converterFactory.registerConverter(new StringToUUIDConverter());
-    this.uacMapperFacade = mapperFactory.getMapperFacade(UAC.class, UniqueAccessCodeDTO.class);
-    this.caseMapperFacade =
-        mapperFactory.getMapperFacade(CollectionCase.class, UniqueAccessCodeDTO.class);
-  }
+  public UniqueAccessCodeServiceImpl() {}
 
   @Override
   public UniqueAccessCodeDTO getAndAuthenticateUAC(String uacHash) throws CTPException {
 
     UniqueAccessCodeDTO data;
     Optional<UAC> uacMatch = dataRepo.readUAC(uacHash);
     if (uacMatch.isPresent()) {
+      // we found UAC
       String caseId = uacMatch.get().getCaseId();
       if (!StringUtils.isEmpty(caseId)) {
+        // UAC has a caseId
         Optional<CollectionCase> caseMatch = dataRepo.readCollectionCase(caseId);
         if (caseMatch.isPresent()) {
+          // Case found
           data = createUniqueAccessCodeDTO(uacMatch.get(), caseMatch, CaseStatus.OK);
         } else {
-          log.warn("Failed to retrieve Case for UAC from storage");
-          data = createUniqueAccessCodeDTO(uacMatch.get(), caseMatch, CaseStatus.NOT_FOUND);
+          // Case NOT found
+          log.with(uacHash)
+              .with(caseId)
+              .error("Cannot find Case for UAC - telling UI unlinked - RM remediation required");
+          data = createUniqueAccessCodeDTO(uacMatch.get(), Optional.empty(), CaseStatus.UNLINKED);
+          data.setCaseId(null);
         }
-        sendRespondentAuthenticatedEvent(data);
       } else {
-        log.warn("Retrieved UAC CaseId not present");
-        data = createUniqueAccessCodeDTO(uacMatch.get(), Optional.empty(), CaseStatus.UNKNOWN);
+        // unlinked
+        data = createUniqueAccessCodeDTO(uacMatch.get(), Optional.empty(), CaseStatus.UNLINKED);
```

Comment:
Is it not worth logging this path?
Would have thought it would help with debugging in Whitelodge, etc

Reply:
have added log debug - unlinked UACS will be as normal as linked ones